### PR TITLE
(SIMP-1897) Open iptables ports from hiera

### PR DESCRIPTION
**Pull request by jeefberkey**
This commit adds iterator to init.pp to read a class parameter, $ports,
that contains a hash of hashes as documented in the class. You can
specifiy this parameter in hiera to open ports up on many boxes easily.

SIMP-608 #comment done iptables
SIMP-1897 #close

### CODE REVIEW

**Comment by jeefberkey** [comment]
See some previous discussion at the old Gerrit change: https://review.gerrithub.io/#/c/302632/

**Comment by jeefberkey** [comment]
# ⚠️ 

Progress on this is halted until January. It's currently not ready for review.

**Comment by trevor-vaughan** [comment]
@lamawithonel You OK with the changes that @jeefberkey made?

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -52,6 +52,26 @@
           it { is_expected.to compile.with_all_deps }
           it { is_expected.to create_iptables_rule('prevent_ipv6_localhost_spoofing').with_apply_to('ipv6') }
         end
+
+        context 'iptables_ports_hash in hiera' do
```
Comment (lamawithonel) [inline spec/classes/init_spec.rb]: Nitpick: Contexts should be labeled so when strung together with their parent `describe` or `context`, and their children expectations, they form a proper sentence.  For example,

```
describe 'foo' do
  context 'when doing a thing' do
    it { is_expected.to return('bar')
  end
end
```

This strings together as `foo when doing a thing is expected to return bar`.

**Pull request merged**